### ISSUE TITLE
Issues with structure serialization

### ISSUE THREAD

**Issue by NSentinel** [dahomey-technologies/Dahomey.Json#1]
Thank you very much for developing a library that eliminates the current limitations of `System.Text.Json`! I wanted to do it myself, but in the process of looking for options, I found your solution.

I've had a few issues using the library at the moment, and since you've already started working in this direction, you might want to solve them.

In the examples below, `JsonHelpers.DefaultOptions.DefaultJsonFormat` includes setting extensions for options.

Using fields in a class works fine, but there are two similar cases with broken ones:

1. The use of primitive fields in the structure:

```csharp
        public struct FieldSerializationStruct
        {
            public int A { get; set; }
            public int B;

            public FieldSerializationStruct (int a, int b)
            {
                A = a;
                B = b;
            }
        }

string jsonSerialized = JsonSerializer.Serialize(new FieldSerializationStruct(1, 2), JsonHelpers.DefaultOptions.DefaultJsonFormat);
```

In this case, field B will be simply skipped

2. A class with a nested structure (a field or property does not matter):

```csharp
        public class FieldSerializationClassWithStruct
        {
            public int A { get; set; } = 1;
            public FieldSerializationStruct B = new FieldSerializationStruct(2,3);
        }

 string jsonSerialized = JsonSerializer.Serialize(new FieldSerializationClassWithStruct(), JsonHelpers.DefaultOptions.DefaultJsonFormat);
```

In this case, an exception occurs:

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Dahomey.Json.Serialization.Converters.MemberConverter`2.Write (Utf8JsonWriter writer, Object obj, JsonSerializerOptions options)
   at Dahomey.Json.Serialization.Converters.ObjectConverter`1.Write (Utf8JsonWriter writer, T value, JsonSerializerOptions options)
   at System.Text.Json.JsonPropertyInfoNotNullable`4.OnWrite (WriteStackFrame & current, Utf8JsonWriter writer)
   at System.Text.Json.JsonPropertyInfo.Write (WriteStack & state, Utf8JsonWriter writer)
   at System.Text.Json.JsonSerializer.Write (Utf8JsonWriter writer, Int32 originalWriterDepth, Int32 flushThreshold, JsonSerializerOptions options, WriteStack & state)
   at System.Text.Json.JsonSerializer.WriteCore (Utf8JsonWriter writer, Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCore (PooledByteBufferWriter output, Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCoreString (Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Serialize [TValue] (TValue value, JsonSerializerOptions options)
```

For "standard" fields like `Guid`, `decimal` it works correctly.

I would also like to note two things:

1. It would be great if the library supported the use of the `DataMember` attribute to set the name  so that you don't have to duplicate attributes with `JsonPropertyName` when using different serializers in the projects.

```csharp
        public class MemberNameSerialization
        {
            [DataMember (Name = "memberName")]
            public int A { get; set; } = 1;
        }
```

2. It would be more convenient to use `SetupExtensions` if it returned `JsonSerializerOptions`

For example, it would be possible to write the initialization in one line:

```csharp
JsonSerializerOptions IndentedJsonSerializerOptions = 
       new JsonSerializerOptions{WriteIndented = true}.SetupExtensions();

```

Thanks in advance!

**Comment by mcatanzariti** [comment]
I will look into you requests!

**Comment by mcatanzariti** [comment]
In the future, it would be easier for me if you could seperate request in different issues. Thank you!

**Comment by NSentinel** [comment]
Thank you very much!

Sure, I will break up in the future.

**Comment by mcatanzariti** [comment]
Implemented in [1.0.9](https://github.com/dahomey-technologies/Dahomey.Json/releases/tag/1.0.9)

**Comment by NSentinel** [comment]
Thank you very much!
